src/test/java/com/device/shop/test/controller/UserControllerTest.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Test
+    public void testCreateUser_DataIntegrityViolationException() throws Exception {
+        User user = new User("Alesia", "Pav", "+380-66-789-12-34", "", "password");
+        when(userService.createUser(any(User.class))).thenThrow(new DataIntegrityViolationException(""));
+
+        mockMvc.perform(post("/users")
+                      .contentType(MediaType.APPLICATION_JSON)
+                      .content(objectMapper.writeValueAsString(user)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void testCreateUser_BadRequestException() throws Exception {
+        User user = new User("Alesia", "Pav", "+380-66-789-12-34", "", "password");
+        when(userService.createUser(any(User.class))).thenThrow(new BadRequestException(""));
+
+        mockMvc.perform(post("/users")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(user)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void testGetUserById() throws Exception {
+        User user = new User();
+        user.setId(10L);
+        user.setName("Alesia");
+
+        when(userService.getUserById(10L)).thenReturn(user);
+
+        mockMvc.perform(get("/{id}", 10L))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value(10))
+                .andExpect(jsonPath("$.name").value("Alesia"));
+    }
+
+    @Test
+    public void testGetUserById_EntityNotFoundException() throws Exception{
+        User user = new User("Alesia", "Pav", "+380-66-789-12-34", "john.doe09@gmail.com", "password");
+
+        doThrow(new EntityNotFoundException()).when(userService).getUserById(anyLong());
+
+        mockMvc.perform(get("/{id}", 1L)
+                       .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+
+
+    @Test
+    public void testGetAllUser() throws Exception {
+        User user1 = new User();
+        user1.setId(1L);
+        user1.setName("John");
+
+        User user2 = new User();
+        user2.setId(2L);
+        user2.setName("Jane");
+
+        List<User> users = Arrays.asList(user1, user2);
+
+        when(userService.getAllUser()).thenReturn(users);
+
+        mockMvc.perform(get("/allUsers"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$[0].id").value(1))
+                .andExpect(jsonPath("$[0].name").value("John"))
+                .andExpect(jsonPath("$[1].id").value(2))
+                .andExpect(jsonPath("$[1].name").value("Jane"));
+    }
+
+
+    @Test
+    public void testUpdateUser() throws Exception {
+        User user = new User();
+        user.setId(10L);
```

Comment:
Move this value `10L` to variable and use variable in all places of this method instead of `10L`